### PR TITLE
fix(engine): avoid reusing vnode.elm unless it is style tag for native

### PR DESCRIPTION
**Pull request by caridy**
## Details

* remove logic to attempt to reuse `vnode.elm` unless it is an HTMLStyleElement for native shadow

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

### CODE REVIEW

File: packages/@lwc/engine/src/3rdparty/snabbdom/types.ts [hunk 1]
```diff
@@ -46,11 +46,15 @@ export interface VElement extends VNode {
     elm: Element | undefined;
     text: undefined;
     key: Key;
+    // TODO: issue #1364 - supporting the ability to provisioned a cloned StyleElement
```
Comment (sfdciuie) [inline packages/@lwc/engine/src/3rdparty/snabbdom/types.ts]:
```suggestion
    // TODO: issue #1364 - supporting the ability to provision a cloned StyleElement
```

**Comment by diervo** [comment]
I don't follow this PR...

File: packages/integration-karma/test/rendering/slot-dont-recycle-vm/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { createElement } from 'lwc';
+import Container from 'x/container';
+
+describe('Component', () => {
```
Comment (ekashida) [inline packages/integration-karma/test/rendering/slot-dont-recycle-vm/index.spec.js]: The name of this file (which is also used as part of the test name) doesn't seem accurate. It should probably be something along the lines of `rendering/elements-are-not-recycled`.

File: packages/integration-karma/test/rendering/slot-dont-recycle-vm/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    it('should not have have an error accessing state.foo', function() {
```
Comment (ekashida) [inline packages/integration-karma/test/rendering/slot-dont-recycle-vm/index.spec.js]: Do we expect an exception to occur if this fix regresses? If so, the test should indicate that using `.toThrow()`.  Also, what is `state.foo` referring to?

File: packages/integration-karma/test/rendering/slot-dont-recycle-vm/index.spec.js (hunk 2 above)
Comment (ekashida) [reply packages/integration-karma/test/rendering/slot-dont-recycle-vm/index.spec.js]: It doesn't look like we need this `describe`.

File: packages/integration-karma/test/rendering/elements-are-not-recycled/x/child/child.html [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<template>
+    <template if:true={open}>
+        <!-- If you remove the outer div, it works -->
```
Comment (ekashida) [inline packages/integration-karma/test/rendering/elements-are-not-recycled/x/child/child.html]: What is "it"?

File: packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    // x-child show/hides the slotted content
+    // Note: There is a difference between native and synthetic shadow on how they handle slotted content:
+    // In native shadow: xSimple will be present and connected, but when not visible, is because dont have any assignedSlot.
+    // In the synthetic shadow however, xSimple will be null and not present in the DOM when not rendered.
+
+    return Promise.resolve()
+        .then(() => {
+            child.open = true;
+
+            return Promise.resolve();
```
Comment (ekashida) [inline packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js]: We don't need to return anything here, do we?

File: packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        })
+        .then(() => {
+            const xSimple = elm.shadowRoot.querySelector('x-simple');
+
+            expect(xSimple).not.toBeNull();
+            expect(xSimple.assignedSlot).not.toBeNull();
+
+            child.open = false;
+
+            return Promise.resolve();
```
Comment (ekashida) [inline packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js]: We don't need to return anything here, do we?

File: packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        })
+        .then(() => {
+            const xSimple = elm.shadowRoot.querySelector('x-simple');
+
+            expect(xSimple === null || xSimple.assignedSlot === null).toBeTruthy();
+            child.open = true;
+
+            return Promise.resolve();
```
Comment (ekashida) [inline packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js]: We don't need to return anything here, do we?

File: packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            expect(xSimple.assignedSlot).not.toBeNull();
+
+            child.open = false;
+
+            return Promise.resolve();
+        })
+        .then(() => {
+            const xSimple = elm.shadowRoot.querySelector('x-simple');
+
+            expect(xSimple === null || xSimple.assignedSlot === null).toBeTruthy();
```
Comment (ekashida) [inline packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js]:
```suggestion
            expect(xSimple).toBeNull();
            expect(xSimple.assignedSlot).toBeNull();
```

File: packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import { createElement } from 'lwc';
+import Container from 'x/container';
+
+it('should not have have an error re-rendering slotted content', function() {
```
Comment (ekashida) [inline packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js]: Where do we expect to see the error if elements were recycled? We should use a `expect().not.toThrow()` there.

File: packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js (hunk 8 above)
Comment (jodarove) [reply packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js]: @ekashida these is not the same, your suggestion is `xSimple === null` **&&** `xSimple.assignedSlot === null`

the code intends for it to be or.

File: packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js (hunk 9 above)
Comment (jodarove) [reply packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js]: i dont think so, just by rendering the component we know it does not throw right?

File: packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js (hunk 5 above)
Comment (jodarove) [reply packages/integration-karma/test/rendering/elements-are-not-recycled/index.spec.js]: i thought we did (at least in jest) but we don't ... removed them

**Pull request merged**